lib/AST/ProtocolConformance.cpp

```diff
@@ -285,6 +287,132 @@ AbstractStorageDecl *ProtocolConformance::getBehaviorDecl() const {
   return getRootNormalConformance()->getBehaviorDecl();
 }
 
+ArrayRef<Requirement> ProtocolConformance::getConditionalRequirements() const {
+  CONFORMANCE_SUBCLASS_DISPATCH(getConditionalRequirements, ());
+}
+
+ArrayRef<Requirement>
+ProtocolConformanceRef::getConditionalRequirements() const {
+  if (isConcrete())
+    return getConcrete()->getConditionalRequirements();
+  else
+    // An abstract conformance is never conditional: any conditionality in the
+    // concrete types that will eventually pass through this at runtime is
+    // completely pre-checked and packaged up.
+    return {};
+}
+
+/// Determine if \c sig can prove \c requirement, meaning that it can deduce T:
+/// Foo or T == U (etc.) with the information it knows. This includes checking
+/// against global state, if any/all of the types in the requirement are
+/// concrete, not type parameters.
+static bool isRequirementSatisfied(GenericSignature *sig,
```

Comment:
I actually moved it out of the GenericSignature on the advice of @DougGregor (you should argue with that guy 😛). In particular, this will be checking global conformances/using the module, but it sounds like GenericSignature wants to remove its connection to that. Maybe it could go on the module decl, or just `swift::isRequirementSatisfied` top-level somewhere?

Reply:
The new `GenericSignatureBuilder::lookupConformance`  should be used to answer these queries.

Reply:
(Future-me still wants the API on `GenericSignature` :))

Reply:
Done.